File: 2/3.py
def main(n, arrA, arrB):
    result = ""
    win = {
        1: 3,
        2: 1,
        3: 2,
    }

    for i in range(n):
        if arrA[i] == arrB[i]:
            result += "D\n"
        elif win[arrA[i]] == arrB[i]:
            result += "A\n"
        else:
            result += "B\n"
    return result


print(main(5, [2, 3, 3, 1, 3], [1, 1, 2, 2, 3]))

File: 2/3.js
// 시간 복잡도: O(n)
// 공간 복잡도: O(n)
function main(n, arrA, arrB) {
  let result = "";
  const win = {
    1: 3,
    2: 1,
    3: 2,
  };

  for (let i = 0; i < n; i++) {
    if (arrA[i] === arrB[i]) {
      result += "D\n";
    } else if (win[arrA[i]] === arrB[i]) {
      result += "A\n";
    } else {
      result += "B\n";
    }
  }

  return result;
}

console.log(main(5, [2, 3, 3, 1, 3], [1, 1, 2, 2, 3]));
